src/+replab/+equi/PhasedMatrixPartition.m

```diff
@@ -0,0 +1,434 @@
+                        ind = blk(1,:) + nR*(blk(2,:)-1);
+                        block(ind) = 0;
+                        phase(ind) = 0;
+                    end
+                else
+                    % not zero
+                    remain = blkInds(1);
+                    for blkInd = blkInds(2:end)
+                        blk = blocks{blkInd};
+                        blocks{blkInd} = zeros(2, 0);
+                        blocks{remain} = horzcat(blocks{remain}, blk);
+                        ind = blk(1,:) + nR*(blk(2,:)-1);
+                        block(ind) = remain;
+                    end
+                end
+            end
+            % renumber by removing empty blocks
+            empty = cellfun(@isempty, blocks);
+            toIndex = cumsum(~empty);
+            toIndex(empty) = 0;
+            fromIndex = find(~empty);
+            blocks = blocks(fromIndex);
+            block(block>0) = toIndex(block(block>0));
+            res = replab.equi.PhasedMatrixPartition(nR, nC, m, phase, block, zeroBlock, blocks);
+        end
+
+        function res = fromPhaseAndBlockIndexMatrices(phaseOrder, phase, block)
+        % Creates a phased matrix partition from a matrix of phases and a matrix of block indices
+            nR = size(phase, 1);
+            nC = size(phase, 2);
+            n = max(block(:));
+            assert(all(unique([0;block(:)])' == 0:n));
+            [I, J] = find(block == 0);
+            zeroBlock = [I(:)'; J(:)'];
+            blocks = cell(1, n);
+            for i = 1:n
+                [I, J] = find(block == i);
+                blocks{i} = [I(:)'; J(:)'];
+            end
+            res = replab.equi.PhasedMatrixPartition(nR, nC, phaseOrder, phase, block, zeroBlock, blocks);
+        end
+
+        function [res, I] = fromIndexMatrix(indexMatrix)
+        % Creates a phased matrix partition from a (possibly signed) index matrix
+        %
+        % The index matrix contains integers. Zeros corresponds to coefficients equal to zero,
+        % i.e. cells of the matrix that do not belong to a block. The indices can be signed,
+        % in which case, indices of opposite signs corresponds to coefficients of opposite value
+        %
+        % If there are negative coefficients, then the result ``phaseOrder`` will be 2, otherwise it is 1.
+        %
+        % Note that `.PhasedMatrixPartition` requires all indices from ``1`` to ``nBlocks` to be present
+        % in the partition, in other words, ``abs(indexMatrix)`` must contain all integers from ``1`` to
+        % ``nBlocks``, where ``nBlocks = max(abs(indexMatrix(:)))``.
+        %
+        % If some indices are not present, this method will prune the corresponding blocks. The second output
+        % argument ``I`` gives, for each index in ``indexMatrix``, either the index in the returned `.PhasedMatrix`
+        % or ``0`` if the index does not appear in ``indexMatrix``.
+        %
+        % Args:
```

Comment:
Great explanation!
Should we include also a short pedagogic example with a matrix including negative indices?

Reply:
Done!